packages/@aws-cdk/aws-apigateway/lib/gateway-response.ts

```diff
@@ -0,0 +1,186 @@
+import { Construct, Resource } from '@aws-cdk/core';
+import { CfnGatewayResponse } from './apigateway.generated';
+import { IRestApi } from './restapi';
+
+/**
+ * Properties for a new gateway response.
+ */
+export interface GatewayResponseProps extends GatewayResponseOptions {
+  /**
+   * Rest api resource to target.
+   */
+  readonly restApi: IRestApi;
+
+  /**
+   * Name of the gateway response.
+   * @default - Auto generated
+   */
+  readonly gatewayResponseName?: string;
+}
+
+/**
+ * Options to add gateway response.
+ */
+export interface GatewayResponseOptions {
+  /**
+   * Response type to associate with gateway response.
+   * @see https://docs.aws.amazon.com/apigateway/latest/developerguide/supported-gateway-response-types.html
+   */
+  readonly type: ResponseType;
+
+  /**
+   * Http status code for response.
+   * @default - standart http status code for the response type.
+   */
+  readonly statusCode?: string;
+
+  /**
+   * Custom path, query and headers for response.
+   * @default - No custom paramter will be added to response.
+   */
+  readonly parameters?: { [key: string]: string };
+
+  /**
+   * Custom templates to get mapped as response.
+   * @default - Default reponse will be mapped
+   */
+  readonly templates?: { [key: string]: string };
+
+}
+
+/**
+ * This resource creates a gateway response which will be applied to response from api gateway
+ * as per the configuration based on the response type.
+ * You will probably want to use `RestApi.addGatewayResponse` for adding new gateway response.
+ *
+ * @resource AWS::ApiGateway::GatewayResponse
+ */
+export class GatewayResponse extends Resource {
+  constructor(scope: Construct, id: string, props: GatewayResponseProps) {
+    super(scope, id, {
+      physicalName: props.gatewayResponseName,
+    });
+
+    const resource = new CfnGatewayResponse(this, 'Resource', {
+      restApiId: props.restApi.restApiId,
+      responseType: props.type,
+      responseParameters: props.parameters,
+      responseTemplates: props.templates,
+      statusCode: props.statusCode,
+    });
+
+    this.node.defaultChild = resource;
+  }
+}
+
+/**
+ * Supported types of gateway responses.
+ */
+export enum ResponseType {
```

Comment:
Let's change this from an enum to a class, with static constants.

This way we can also encode a `custom()` option, that allows for flexibility for when a new type is added, but support is not yet available in the CDK.

See a similar pattern here - https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-cognito/lib/user-pool.ts#L143